drivers/Ticker.h

```diff
@@ -75,12 +75,17 @@ class Ticker : public TimerEvent, private NonCopyable<Ticker> {
 
     /** Attach a function to be called by the Ticker, specifying the interval in seconds
      *
+     *  The method forwards its arguments to attach_us() rather than copying them which
+     *  may not be trivial depending on the callback copied.
+     *  The function is forcibly inlined to not use floating-point operations. This is
+     *  possible given attach_us() expects an integer value for the callback interval.
      *  @param func pointer to the function to be called
      *  @param t the time between calls in seconds
      */
-    void attach(Callback<void()> func, float t)
+    template <typename F>
+    MBED_FORCEINLINE void attach(F &&func, float t)
     {
-        attach_us(func, t * 1000000.0f);
+        attach_us(std::forward<F>(func), t * 1000000.0f);
```

Comment:
You should put the include for `forward` in.

ARMC5 compatibility needs it to be `#include <mstd_utility>`, rather than just `#include <utility>`

(See https://github.com/ARMmbed/mbed-os/blob/master/platform/cxxsupport/README.md - I need to get on with the handbook page for that).

Reply:
Some points first:
1. Without using what you are suggesting above, the code compiles fine for ARMC5.
2. With or without what you are suggesting there is still one floating point instruction (only for ARMC5, have not looked at where it comes from)

Now the question:
1. Is it worth doing given we are moving to ARMC6?
2. Are you suggesting adding 2 conditional pre-processor directives in Ticker.h as follows?
```
#if defined(__CC_ARM)
#include <mstd_utility>
#endif
...

    template <typename F>
    MBED_FORCEINLINE void attach(F &&func, float t)
    {
        attach_us(
#if defined(__CC_ARM)
            mstd::forward<F>(func),
#else
            std::forward<F>(func),
#endif // defined(__CC_ARM)
            t * 1000000.0f
        );
    }
```

Reply:
1. We're trying not to break the ARMC5 build outright for now, which is what that cxxsupport directory is partly about. It should be minimal effort from most people - I'm taking the hit. There's no need for `#ifdef`s anywhere in code - all the ifdeffing happens in the cxxsupport headers, but you just need to make sure to use `<mstd_utility>`, to give us the chance to get in front of the ARMC5's built-in `<utility>`

1. No, the only change is to add `#include <mstd_utility>`, unconditionally. That works for all toolchains. You can keep it as `std::forward`.

Reply:
d4497b7